frontend/src/lib/lemon-ui/Tooltip/Tooltip.tsx

```diff
@@ -1,61 +1,140 @@
+        placement,
+        open,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offsetFunc(offset),
+            flip({ fallbackAxisSideDirection: 'start' }),
+            shift(),
+            arrow({ element: caretRef }),
+        ],
+    })
+
+    const hover = useHover(context, {
+        move: false,
+        delay: {
+            open: delayMs,
+            close: 0,
+        },
+    })
+    const focus = useFocus(context)
+    const dismiss = useDismiss(context)
+    const role = useRole(context, { role: 'tooltip' })
+
+    const { getFloatingProps, getReferenceProps } = useInteractions([hover, focus, dismiss, role])
+
+    const { styles: transitionStyles } = useTransitionStyles(context, {
+        duration: {
+            open: 200,
+            close: 0,
+        },
+        initial: ({ side }) => ({
+            opacity: 0,
+            transform: {
+                top: 'translateY(3px)',
+                bottom: 'translateY(-3px)',
+                left: 'translateX(3px)',
+                right: 'translateX(-3px)',
+            }[side],
+        }),
+    })
+
+    const childrenRef = (children as any).ref
+    const triggerRef = useMergeRefs([refs.setReference, childrenRef])
+
     const child = React.isValidElement(children) ? children : <span>{children}</span>
 
-    const derivedVisible = typeof visible === 'undefined' ? localVisible && debouncedLocalVisible : visible
-
-    return props.title ? (
-        <AntdTooltip
-            {...props}
-            getPopupContainer={floatingContainer ? () => floatingContainer : undefined}
-            visible={derivedVisible}
-        >
-            {React.cloneElement(child, {
-                onMouseEnter: () => {
-                    child.props.onMouseEnter?.()
-                    if (typeof visible === 'undefined') {
-                        setVisible(true)
-                    }
-                },
-                onMouseLeave: () => {
-                    child.props.onMouseLeave?.()
-                    if (typeof visible === 'undefined') {
-                        setVisible(false)
-                    }
-                },
-            })}
-        </AntdTooltip>
+    const clonedChild = React.cloneElement(
+        child,
+        getReferenceProps({
+            ref: triggerRef,
+            ...child.props,
+        })
+    )
+
+    return title ? (
+        <>
+            {clonedChild}
+            {open && (
+                <FloatingPortal>
+                    <div
+                        ref={refs.setFloating}
+                        // eslint-disable-next-line react/forbid-dom-props
+                        style={{ ...context.floatingStyles }}
+                        {...getFloatingProps()}
+                    >
+                        <div
+                            className={clsx(
+                                'bg-tooltip-bg py-1.5 px-2 z-[1070] break-words rounded text-start text-white',
```

Comment:
Only the `z-[1070]` bit is tricky, because it's hard to keep track of all the z-indexes being used. For this one thing I think using a CSS var with `style` would be better for long-term maintainability, thanks to having the source of truth in `vars.scss`

Reply:
Good call. ✅ to the variable, what do you think about `z-[var(--z-tooltip)]` as a class rather than using the `style` prop? I love that it works but conscious it's a new pattern and is somewhat difficult read instantly

Reply:
I needed a class for the `box-shadow` as it turns out so created the CSS file